y3prediction/prediction.py

```diff
@@ -511,18 +512,56 @@ def main(actx_class,
     mesh_filename = configurate("mesh_filename", input_data, "data/actii_2d.msh")
     noslip = configurate("noslip", input_data, True)
-    adiabatic = configurate("adiabatic", input_data, False)
     use_1d_part = configurate("use_1d_part", input_data, True)
 
-    # setting these in the input file toggles the check off
-    # support for legacy runs only where you could specify boundary tags
-    # that were unused in certain cases
-    use_outflow_boundary = configurate("use_outflow_boundary", input_data, False)
-    use_inflow_boundary = configurate("use_inflow_boundary", input_data, False)
-    use_flow_boundary = configurate("use_flow_boundary", input_data, True)
-    use_injection_boundary = configurate("use_injection_boundary", input_data, True)
-    use_wall_boundary = configurate("use_wall_boundary", input_data, True)
-    use_interface_boundary = configurate("use_interface_boundary", input_data, True)
+    # setting these to none in the input file toggles the check for that
+    # boundary off provides support for legacy runs only where you could
+    # specify boundary tags that were unused in certain cases
+    use_outflow_boundary = configurate(
+        "use_outflow_boundary", input_data, "none")
+    use_inflow_boundary = configurate(
+        "use_inflow_boundary", input_data, "none")
+    use_flow_boundary = configurate(
+        "use_flow_boundary", input_data, "prescribed")
+    use_injection_boundary = configurate(
+        "use_injection_boundary", input_data, "none")
+    use_wall_boundary = configurate(
+        "use_wall_boundary", input_data, "isothermal_noslip")
+    use_interface_boundary = configurate(
+        "use_interface_boundary", input_data, "none")
+
+    # for each tagged boundary surface, what are they assigned to be
+    # isothermal wall -> wall when current running simulation support is not needed
+    #
+    # wall_interface is automatically generated from a shared fluid/solid volume
+    # interface and only when the solid volume is turned off by use_wall
+    bndry_config = {"outflow": use_outflow_boundary,
+                    "inflow": use_inflow_boundary,
+                    "flow": use_flow_boundary,
+                    "injection": use_injection_boundary,
+                    "isothermal_wall": use_wall_boundary,
+                    "wall_interface": use_interface_boundary}
+
+    # list of strings that are allowed to defined boundary conditions
+    allowed_boundary_types = [
+        "none",
+        "isothermal_noslip",
+        "isothermal_slip",
+        "adiabatic_noslip",
+        "adiabatic_slip",
+        "prescribed"
+    ]
+
+    # boundary sanity check
+    def boundary_type_sanity(boundary, boundary_type):
+        if boundary_type not in allowed_boundary_types:
+            error_message = ("Invalid boundary specification "
+                             f"{boundary_type} for {boundary}")
+            if rank == 0:
+                raise RuntimeError(error_message)
```

Comment:
After importing from simutils at the top.

```suggestion
            raise SimulationConfigurationError(error_message)
```